multinet/auth/google.py

```diff
@@ -0,0 +1,159 @@
+"""Handling of Google Authorization."""
+import requests
+import base64
+import json
+
+from os import getenv
+from flask import Flask, redirect, request, make_response, url_for
+from werkzeug.wrappers import Response as ResponseWrapper
+from flask.blueprints import Blueprint
+from authlib.integrations.flask_client import OAuth
+
+from webargs.flaskparser import use_kwargs
+from webargs import fields
+
+from multinet.user import (
+    load_user_from_cookie,
+    load_user,
+    updated_user,
+    get_user_cookie,
+    set_user_cookie,
+    register_user,
+    filter_user_info,
+    filter_document_meta,
+)
+
+from multinet.auth.types import GoogleUserInfo, User
+
+from typing import Dict
+
+
+CLIENT_ID = getenv("GOOGLE_CLIENT_ID")
+CLIENT_SECRET = getenv("GOOGLE_CLIENT_SECRET")
+
+GOOGLE_BASE_API = "https://www.googleapis.com/"
+GOOGLE_USER_INFO_URL = "oauth2/v3/userinfo"
+
+MULTINET_COOKIE = "multinet-token"
+
+bp = Blueprint("google", "google")
+oauth = OAuth()
+
+states_to_return_urls = {}
+
+
+def parse_id_token(token: str) -> GoogleUserInfo:
+    """Parse the base64 encoded id token."""
+    parts = token.split(".")
+    if len(parts) != 3:
+        # Should probably throw something here
+        pass
+
+    payload = parts[1]
+    padded = payload + ("=" * (4 - len(payload) % 4))
```

Comment:
Just curious, why do they make you pad this value before decoding it?

Reply:
I can't find a clear reason why, it's just defined in the spec :man_shrugging: